GeneratorInterface/RivetInterface/plugins/HTXSRivetProducer.cc

```diff
@@ -108,13 +105,15 @@ void HTXSRivetProducer::produce(edm::Event& iEvent, const edm::EventSetup&) {
         } else if (nBs == 2 && nHs == 1 && nZs == 0) {
           m_HiggsProdMode = HTXS::BBH;
         }
-
-        _HTXS->setHiggsProdMode(m_HiggsProdMode);
       }
     }
 
-    if (_isFirstEvent) {
-      _analysisHandler.addAnalysis(_HTXS);
+    if (_isFirstEvent || !_HTXS->hasProjection("FS")) {
```

Comment:
`_isFirstEvent` does not seem to be needed anymore, instead you could just test `_HTXS` is not `nullptr`.